### ISSUE TITLE
realm 3.1.0 error 

### ISSUE THREAD

**Issue by MuratAkbaba006** [realm/realm-js#5169]
### How frequently does the bug occur?

All the time

### Description

I have a problem with realm version 3.1.0.  

One of my old applications was working before but I had to make a little change and wanted to generate a new apk.
Even I didn't make any changes on realm or any other library, I'm facing with this problem that you can see below. 

### Stacktrace & log output

```shell
error [Error: Realm object constructor must not return another value]
```

### Can you reproduce the bug?

Yes, always

### Version

realm 3.1.0

### What SDK flavour are you using?

Local Database only

### Are you using encryption?

No, not using encryption

### Platform OS and version(s)

mac monterey 12.6 

### Build environment

Which debugger for React Native: ..

**Comment by kneth** [comment]
Version 3.1.0 was released September 19, 2019 and it great to greet an old, loyal user :smile:

It is worth verifying that `npm` (or `yarn`) hasn't silently upgraded Realm. You can check the version in `node_modules/realm/package.json`. 

Do you know what line in your app, the error message is coming from? Is it possible to share it?

**Comment by MuratAkbaba006** [comment]
Yes its quite old i know but it is really hard to update whole package for me now, i guess it is too late :D

At the directory you said, version is shown as 3.6.0 

The usage is as shown below, I'm not the one who wrote it actually so, i might be mistaken.
UserProfile.create(userInfo, true);

<img width="459" alt="MicrosoftTeams-image (13)" src="https://user-images.githubusercontent.com/44754934/206455075-4deac871-0f46-44fd-bd0e-2b1ea80517c3.png">
<img width="459" alt="MicrosoftTeams-image (12)" src="https://user-images.githubusercontent.com/44754934/206455163-e4817ab0-d99f-49dc-ad52-7bc14def1ebd.png">

**Comment by kneth** [comment]
You could try to pin the Realm version to 3.1.0 in `package.json`:

```
  "realm": "3.1.0",
```

See https://docs.npmjs.com/cli/v9/configuring-npm/package-json#dependencies for how to specify version

**Comment by MuratAkbaba006** [comment]
I fixed the problem by upgrading to 5.0.3. I had to downgrade node version to 10.x.x btw. Thanks for the responses

**Comment by kneth** [comment]
@MuratAkbaba006 Good to hear. I hope you in the future will be able to upgrade to a more recent version.
